File: backend/routes/quiz.py
from flask import Blueprint, request, jsonify
import random

quiz_bp = Blueprint('quiz', __name__)

# Sample quiz questions
questions = [
    {
        "question": "What is the area of a rectangle with length 5 and width 3?",
        "options": [15, 8, 20, 18],
        "answer": 15
    },
    {
        "question": "What is the area of a circle with radius 7? (Use π ≈ 3.14)",
        "options": [153.86, 140.0, 154.0, 160.0],
        "answer": 153.86
    },
    {
        "question": "What is the area of a triangle with base 4 and height 5?",
        "options": [10, 12, 15, 20],
        "answer": 10
    }
]

@quiz_bp.route('/quiz/questions', methods=['GET'])
def get_questions():
    return jsonify(questions)

@quiz_bp.route('/quiz/submit', methods=['POST'])
def submit_quiz():
    data = request.json
    score = 0
    for question, user_answer in zip(questions, data['answers']):
        if question['answer'] == user_answer:
            score += 1
    return jsonify({"score": score, "total": len(questions)})

File: frontend/src/js/quiz.js
const quizData = [
    {
        question: "What is the area of a rectangle with a length of 5 and a width of 3?",
        a: "15",
        b: "8",
        c: "18",
        d: "20",
        correct: "a"
    },
    {
        question: "What is the area of a triangle with a base of 4 and a height of 5?",
        a: "10",
        b: "12",
        c: "15",
        d: "20",
        correct: "a"
    },
    {
        question: "What is the area of a circle with a radius of 3? (Use π ≈ 3.14)",
        a: "28.26",
        b: "31.42",
        c: "9.42",
        d: "18.84",
        correct: "a"
    },
    {
        question: "What is the area of a square with a side length of 4?",
        a: "12",
        b: "16",
        c: "20",
        d: "24",
        correct: "b"
    }
];

const quizContainer = document.getElementById('quiz');
const resultContainer = document.getElementById('result');
const submitButton = document.getElementById('submit');

function loadQuiz() {
    quizData.forEach((currentQuiz, index) => {
        const questionElement = document.createElement('div');
        questionElement.classList.add('question');
        questionElement.innerHTML = `
            <h2>${currentQuiz.question}</h2>
            <label>
                <input type="radio" name="question${index}" value="a">
                ${currentQuiz.a}
            </label>
            <label>
                <input type="radio" name="question${index}" value="b">
                ${currentQuiz.b}
            </label>
            <label>
                <input type="radio" name="question${index}" value="c">
                ${currentQuiz.c}
            </label>
            <label>
                <input type="radio" name="question${index}" value="d">
                ${currentQuiz.d}
            </label>
        `;
        quizContainer.appendChild(questionElement);
    });
}

function calculateScore() {
    let score = 0;
    quizData.forEach((currentQuiz, index) => {
        const answer = document.querySelector(`input[name="question${index}"]:checked`);
        if (answer && answer.value === currentQuiz.correct) {
            score++;
        }
    });
    return score;
}

submitButton.addEventListener('click', () => {
    const score = calculateScore();
    resultContainer.innerHTML = `You scored ${score} out of ${quizData.length}`;
});

loadQuiz();